askama_derive/src/generator.rs

```diff
@@ -1053,6 +1074,33 @@ impl<'a> Generator<'a> {
         Ok(size_hint)
     }
 
+    fn handle_embed(
+        &mut self,
+        buf: &mut Buffer,
+        embed: &'a Embed<'_>,
+    ) -> Result<usize, CompileError> {
+        self.flush_ws(embed.ws1);
+        self.write_buf_writable(buf)?;
+        let embed_path = self
+            .input
+            .config
+            .find_template(embed.path, Some(&self.input.path))?;
+        let mut embedded_context =
+            Context::new(self.input.config, &self.input.path, embed.nodes.as_slice())?;
```

Comment:
Probably makes sense to add an additional parameter to set whether or not this context starts at the top level, just so we get an error.

Adding an `{% extends ... %}` (or any top-level only node) inside the embed node is currently valid, even though it doesn't end up doing anything.